### PR TITLE
Adding PR template

### PR DESCRIPTION
**Pull request by joperezr**
Adding basic PR template with a checklist to ensure that we maintain `main` branch ship-ready.

cc: @adityamandaleeka @mitchdenny @eerhardt 
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/5139)

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  - [ ] No
+- Are you including unit tests for the changes and scenario tests if relevant?
+  - [ ] Yes
+  - [ ] No
+- Did you add public API? Did you have an API Review for it?
+  - [ ] Yes
+  - [ ] No
+- Does the change make any security assumptions or guarantees? Have you done a threat model and had a security review?
+  - [ ] Yes
+  - [ ] No
```
Comment (eerhardt) [inline .github/pull_request_template.md:20]: For these 2 lines, they both have 2 questions and the answer can be different for each question. How would you answer that this changes public API but doesn't have an API Review (yet)?

Comment (joperezr) [reply .github/pull_request_template.md:20]: Good feedback, I'll fix this.

**Comment by RussKie** [comment]
It'd be great to see a preview of this template. I tried creating a PR against your fork but because the template isn't in your main it didn't work...

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.
+
+Fixes # (issue)
+
+## Checklist
+
+- Is this feature complete or do we expect follow-up changes before this code can ship?
+  - [ ] Yes
+  - [ ] No
```
Comment (JamesNK) [inline .github/pull_request_template.md:11]: This is a confusing Yes/No question. What are you saying yes to?
```suggestion
- Is this feature complete?
  - [ ] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
```

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+- Are you including unit tests for the changes and scenario tests if relevant?
+  - [ ] Yes
+  - [ ] No
+- Did you add public API?
```
Comment (mitchdenny) [inline .github/pull_request_template.md:15]: Can we also ask whether the public API has XML doc comments including:

1. `<remarks />` if greater explanation is required than what can be provided in the summary.
2. `<example />` elements for any examples with code placed in a `<code />` element.

**Comment by mitchdenny** [comment]
This is a good starting point. Added a comment for some things that I routinely look for.

**Review by mitchdenny — APPROVED** [review]
Approving because this is better than what we have now and we can iterate.

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  - [ ] Yes
+    - If yes, did you have an API Review for it?
+      - [ ] Yes
+      - [ ] No
+  - [ ] No
```
Comment (joperezr) [inline .github/pull_request_template.md:20]:
```suggestion
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [ ] No
  - [ ] No
```

**Pull request merged**